src/main.rs

```diff
@@ -12,11 +12,20 @@ mod utils;
 
 fn main() -> color_eyre::Result<()> {
     // We use an unused var because we only parse for `-h|--help` and `-V|--version`
-    let _args = Cli::parse();
+    let args = Cli::parse();
 
     utils::install_hooks()?;
     let mut terminal = utils::init()?;
     let mut app = App::new();
+
+    if args.show_configs {
+        Logger::info("Printing current configurations");
+        drop(terminal);
+        utils::restore()?;
+        println!("{}", serde_json::to_string_pretty(&app.config)?);
+        return Ok(());
+    }
```

Comment:
Couldn't we move this part to `src/cli.rs`? In the sense of centralizing the handling of CLI options? I understand we need to do this particularly after `App::new()` and that this won't be the case for most of other options (if any), but the point is that we should keep `main.rs` as lean as possible.

Reply:
Ok moved it to a `Cli::resolve` method that can be modified when we add new args